src/last_sign_state.rs

```diff
@@ -0,0 +1,200 @@
+use abscissa::Error;
+use serde_json;
+use std::{
+    fmt::{self, Display},
+    fs::File,
+    io::prelude::*,
+    path::Path,
+};
+use tendermint::{block, chain};
+
+#[derive(Serialize, Deserialize)]
+struct LastSignData {
+    pub height: i64,
+    pub round: i64,
+    pub step: i8,
```

Comment:
Can these ever actually be negative?

Reply:
Not entirely sure. I have a vague memory that in Tendermint sometimes these values get set at -1 as a marker.

But I haven't actually looked through all the code.

Reply:
Also those are the types in the Tendermint implementation.

Reply:
As far as I know the only thing that can be negative is the Proposal.POLRound (can be -1 as @zmanian mentioned). The other fields can not be negative. But yeah the types are all signed in the tendermint golang code ... Not sure we need to follow this tho.

Reply:
I might suggest using unsigned integers for anything that isn't explicitly negative, and potentially `Option` in lieu of `-1`